athena-vertica/src/main/java/com/amazonaws/connectors/athena/vertica/VerticaSchemaUtils.java

```diff
@@ -0,0 +1,109 @@
+/*-
+ * #%L
+ * athena-vertica
+ * %%
+ * Copyright (C) 2019 - 2020 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.connectors.athena.vertica;
+
+import com.amazonaws.athena.connector.lambda.data.SchemaBuilder;
+import com.amazonaws.athena.connector.lambda.domain.TableName;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class VerticaSchemaUtils
+{
+    private static final Logger logger = LoggerFactory.getLogger(VerticaSchemaUtils.class);
+
+    //Builds the table schema
+    protected Schema buildTableSchema(Connection connection, TableName name)
+    {
+        try
+        {
+            logger.info("Building the schema for table {} ",name);
+            SchemaBuilder tableSchemaBuilder = SchemaBuilder.newBuilder();
+            DatabaseMetaData dbMetadata = connection.getMetaData();
+            ResultSet definition = dbMetadata.getColumns(null, name.getSchemaName(), name.getTableName(), null);
+            while(definition.next())
+            {
+                //todo: review this again and add the Vertica data types missed
+                //If Bit
+                if (definition.getString("TYPE_NAME").equalsIgnoreCase("BIT"))
```

Comment:
can we turn this into if/else or use continue; to skip the rest of the checks once we find one that matches? otherwise this is going to check a bunch of things needlessly.

Reply:
converted to case statement